src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -727,6 +727,76 @@ void CVulkanLogicalDevice::updateDescriptorSets_impl(const SUpdateDescriptorSets
     m_devf.vk.vkUpdateDescriptorSets(m_vkdev,vk_writeDescriptorSets.size(),vk_writeDescriptorSets.data(),vk_copyDescriptorSets.size(),vk_copyDescriptorSets.data());
 }
 
+void CVulkanLogicalDevice::nullifyDescriptors_impl(const std::span<const IGPUDescriptorSet::SDropDescriptorSet> drops, asset::IDescriptor::E_TYPE descriptorType)
+{
+    if (getEnabledFeatures().nullDescriptor)
+    {
+		core::vector<VkWriteDescriptorSet> vk_writeDescriptorSets(drops.size(),{VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET,nullptr});
+		core::vector<VkWriteDescriptorSetAccelerationStructureKHR> vk_writeDescriptorSetAS(69u,{VK_STRUCTURE_TYPE_WRITE_DESCRIPTOR_SET_ACCELERATION_STRUCTURE_KHR,nullptr});
+
+        size_t maxSize = 0;
+        for (auto i = 0; i < drops.size(); i++)
+        {
+            const auto& write = drops[i];
+            maxSize = core::max(maxSize, write.count);
+        }
+        size_t descriptorSize;
+        switch (asset::IDescriptor::GetTypeCategory(descriptorType))
+        {
+			case asset::IDescriptor::EC_BUFFER:
+                descriptorSize = sizeof(VkDescriptorBufferInfo);
+                break;
+			case asset::IDescriptor::EC_IMAGE:
+                descriptorSize = sizeof(VkDescriptorImageInfo);
+                break;
+			case asset::IDescriptor::EC_BUFFER_VIEW:
+                descriptorSize = sizeof(VkBufferView);
+                break;
+			case asset::IDescriptor::EC_ACCELERATION_STRUCTURE:
+                descriptorSize = sizeof(VkAccelerationStructureKHR);
```

Comment:
btw if you look athe the  validation params for copies/updates they gather these counts in advance during validation